### ISSUE TITLE
IIncrementalGenerator vs ISourceGenerator

### ISSUE THREAD

**Issue by ignatandrei** [NeVeSpl/NSourceGenerators#1]
I maintain a list of Roslyn Source Code Generators at [ignatandrei.github.io/RSCG_Examples/v2/docs/List-of-RSCG](https://ignatandrei.github.io/RSCG_Examples/v2/docs/List-of-RSCG)

However, the source generators have been deprecated in favor of the new incremental generators , that allows better performance

See [dotnet/roslyn@main/docs/features/incremental-generators.md](https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md?rgh-link-date=2024-10-09T08%3A55%3A53Z)

However this means passing from ISourceGenerator to IIncrementalGenerator

Could you do that ?

Thanks
Andrei

**Comment by NeVeSpl** [comment]
As far as I remember, incremental generators do not work with .net framework projects. I still need this source generator to be compatible with legacy projects thus changing it to incremental is not an option right now.

**Comment by ignatandrei** [comment]
I do not think that is such a limitation. What version of .NET Framework are you thinking about?

**Comment by NeVeSpl** [comment]
.net framework 4.8

**Comment by ignatandrei** [comment]
I have tested with one of my generators and the console project
    <TargetFrameworkVersion>v4.7.2</TargetFrameworkVersion>

It works.
